wptrunner/executors/pytestrunner/fixtures.py

```diff
@@ -45,14 +46,79 @@ def setup(request, session):
         def test_something(setup, session):
             assert session.url == "https://example.org"
 
-    The session is closed when the test module goes out of scope by an
-    implicit call to `session.end`.
+    When the test function goes out of scope, any remaining user prompts
+    and opened windows are closed, and the current browsing context is
+    switched back to the top-level browsing context.
+
+    After returning the session to a near-pristine state, the session
+    is also closed by an implicit call to `session.end`.
     """
 
     def __init__(self, client):
         self.client = client
 
-    @pytest.fixture(scope="module")
+        # store handle of original window
+        # so that we do not close it
+        self.original_window = self.client.window_handle
+        self.client.end()
```

Comment:
@lukeis Sorry for my late response to this PR!

So I incorporated some of the suggestions you made, but instead of explicitly forcing a new session for each test (which would be expensive), I let it dismiss user prompts, restore windows, and switch to the top-level browsing context (in that order) after each test function.

Does that sound acceptable?